src/ServiceMap.php

```diff
@@ -43,9 +50,35 @@ public function __construct(string $containerXml)
 		}
 	}
 
-	public function getServices(): array
+	/**
+	 * @param mixed $node
+	 * @return array|null
+	 */
+	public function getServiceFromNode($node)
+	{
+		$serviceId = self::getServiceIdFromNode($node);
+
+		if ($serviceId !== null && \array_key_exists($serviceId, $this->services) && !$this->services[$serviceId]['synthetic']) {
+			return $this->services[$serviceId];
+		}
+
+		return null;
+	}
+
+	/**
+	 * @param mixed $node
+	 * @return null|string
+	 */
+	public static function getServiceIdFromNode($node)
```

Comment:
I'm not a fan of having this here... but I needed a way to get the ID there : https://github.com/hectorj/phpstan-symfony/blob/ba8dc8d950a5da6117172f1e231fe55b75f34715/src/Rules/ContainerInterfaceUnknownServiceRule.php#L38

I'm open to suggestions.
Don't hesitate to edit the PR if you want to refactor.